libs/vkd3d/command.c

```diff
@@ -5041,19 +5155,31 @@ static void STDMETHODCALLTYPE d3d12_command_list_ResourceBarrier(d3d12_command_l
             d3d12_command_list_track_resource_usage(list, resource);
     }
 
+    /* Normalize pipeline stage / access flags. */
+    if (src_stage_mask & VK_PIPELINE_STAGE_ALL_COMMANDS_BIT)
+        src_stage_mask = VK_PIPELINE_STAGE_ALL_COMMANDS_BIT;
+    if (dst_stage_mask & VK_PIPELINE_STAGE_ALL_COMMANDS_BIT)
+        dst_stage_mask = VK_PIPELINE_STAGE_ALL_COMMANDS_BIT;
+    if (vk_memory_barrier.srcAccessMask & VK_ACCESS_MEMORY_WRITE_BIT)
+        vk_memory_barrier.srcAccessMask = VK_ACCESS_MEMORY_WRITE_BIT;
+    if (vk_memory_barrier.dstAccessMask & VK_ACCESS_MEMORY_WRITE_BIT)
+        vk_memory_barrier.dstAccessMask = VK_ACCESS_MEMORY_READ_BIT | VK_ACCESS_MEMORY_WRITE_BIT;
```

Comment:
RADV doesn't support `VK_ACCESS_MEMORY_*_BIT`, so we'd break existing barriers by using these.

In general I wonder if we really need to do the normalization at all, it might not be pretty to have something like `ALL_COMMANDS | TRANSFER` but it should be legal and work fine regardless.

Reply:
That's a ... particular driver bug. Spec is very clear on this. Is there a bug report for this?

Reply:
Not that I'm aware of; I stumbled across that a while ago but forgot about it.

Reply:
Fixed.